### PR TITLE
GDPR Export

### PR DESCRIPTION
**Pull request by kobergj**
Adds an endpoint to get all data related to a user

### CODE REVIEW

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/20947/68/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20947/tracing/space-members-can-be-invited-with-an-expiration-date-alice-2023-3-30-08-23-40.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20947/tracing/space-members-can-be-invited-with-an-expiration-date-brian-2023-3-30-08-23-47.zip 


</p></details>

File: services/graph/pkg/service/v0/personaldata.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	if err := createFolders(ctx, ref, gwc); err != nil {
+		return err
+	}
+
+	return touchFile(ctx, ref, gwc)
+}
+
+func touchFile(ctx context.Context, ref *provider.Reference, gwc gateway.GatewayAPIClient) error {
+	resp, err := gwc.TouchFile(ctx, &provider.TouchFileRequest{
+		Opaque: utils.AppendPlainToOpaque(nil, "markprocessing", "true"),
```
Comment (micbar) [inline services/graph/pkg/service/v0/personaldata.go:168]: I assume the reva update is already part of that PR

File: services/graph/pkg/service/v0/personaldata.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+		return fmt.Errorf("unexpected statuscode while touching file: %d %s", resp.GetStatus().GetCode(), resp.GetStatus().GetMessage())
+	}
+	return nil
+}
+
+func createFolders(ctx context.Context, ref *provider.Reference, gwc gateway.GatewayAPIClient) error {
+	var pathes []string
+	p := filepath.Dir(ref.GetPath())
+	for p != "." {
+		pathes = append([]string{p}, pathes...)
```
Comment (ainmosni) [inline services/graph/pkg/service/v0/personaldata.go:184]:
```suggestion
		paths = append([]string{p}, paths...)
```

File: services/graph/pkg/service/v0/personaldata.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+		p = filepath.Dir(p)
+	}
+
+	for _, p := range pathes {
```
Comment (ainmosni) [inline services/graph/pkg/service/v0/personaldata.go:188]:
```suggestion
	for _, p := range paths {
```

**Comment by micbar** [comment]
Looks good in general.

open items:

[ ] Add Request to the libregraph spec

File: services/graph/pkg/service/v0/personaldata.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+		return err
+	}
+	if resp.GetStatus().GetCode() != rpc.Code_CODE_OK {
+		return fmt.Errorf("unexpected statuscode while touching file: %d %s", resp.GetStatus().GetCode(), resp.GetStatus().GetMessage())
+	}
+	return nil
+}
+
+func createFolders(ctx context.Context, ref *provider.Reference, gwc gateway.GatewayAPIClient) error {
+	var pathes []string
```
Comment (ainmosni) [inline services/graph/pkg/service/v0/personaldata.go:181]:
```suggestion
	var paths []string
```

**Review by ainmosni — CHANGES_REQUESTED** [review]
Just found a small typo.

**Review by ainmosni — COMMENTED** [review]
Just found a small typo.

File: services/graph/pkg/service/v0/personaldata.go (hunk 1 above)
Comment (kobergj) [reply services/graph/pkg/service/v0/personaldata.go:168]: Yes. Already part of this PR

**Review by micbar — APPROVED** [review]

**Pull request merged**